core/unix/signal.c

```diff
@@ -2676,12 +2675,16 @@ thread_set_self_context(void *cxt)
     memset(&frame, 0, sizeof(frame));
 #ifdef LINUX
 #    ifdef X86
+    dcontext_t *dcontext = get_thread_private_dcontext();
     byte *xstate = get_and_initialize_xstate_buffer(dcontext);
     frame.uc.uc_mcontext.fpstate = &((kernel_xstate_t *)xstate)->fpstate;
 #    endif /* X86 */
     frame.uc.uc_mcontext = *sc;
 #endif
+    IF_ARM(ASSERT_NOT_TESTED());
+#if defined(LINUX) && defined(X86)
```

Comment:
@AssadHashmi Do we need any special handling for fpstate on AArch64/ARM?

Copying over the [conversation](https://github.com/DynamoRIO/dynamorio/issues/4496#issuecomment-718945863) from #4496. 

> > abhinav92003: @derekbruening About the `ASSERT_NOT_IMPLEMENTED` in `save_fpstate` for AArch64: Do we need special handling for fpstate on AArch64/ARM?
> 
> derekbruening: I don't think so: dr_mcontext.simd[] should hold all the direct values (i.e., it's not like x86 where the x87 state is not in dr_mcontext_t _at all_). Maybe there are status registers or sthg though? I remember this coming up recently in some issue or PR and Assad was not sure -- I can't find the issue though: recent ones are #4467, #4488.
> 
> @AssadHashmi -- if you could comment.

Reply:
Apologies for the late response @abhinav92003 and @derekbruening, this slipped under the mat.

> Do we need special handling for fpstate on AArch64/ARM?

Not in this case AFAICT.

For future reference, the full implementation of `save_fpstate` for AArch64 is tracked by https://github.com/DynamoRIO/dynamorio/issues/2629

Reply:
Thanks Assad and Derek. (Some more discussion on #2629).